<!doctype html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta http-equiv="last-modified" content="2025-11-22 10:15:12 +0000">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <!-- meta "search-domain" used for google site search function google_search() -->
    <meta name="search-domain" value="">
    <link rel="stylesheet" type="text/css" href="../assets/css/bootstrap.css" />
    <link rel="stylesheet" type="text/css" href="../assets/css/bootstrap-theme.css" />
    <link rel="stylesheet" type="text/css" href="../assets/css/lesson.css" />
    
    <!-- HTML5 shim and Respond.js for IE8 support of HTML5 elements and media queries -->
    <!-- WARNING: Respond.js doesn't work if you view the page via file:// -->
    <!--[if lt IE 9]>
	<script src="https://oss.maxcdn.com/html5shiv/3.7.2/html5shiv.min.js"></script>
	<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
	<![endif]-->
    <title>Quantitative Trait Mapping: Estimated QTL effects</title>
  </head>
  <body>
    <div class="container">
      
<nav class="navbar navbar-default">
  <div class="container-fluid">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#bs-example-navbar-collapse-1" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>

      
      

      
      <a class="navbar-brand" href="../">Home</a>

    </div>
    <div class="collapse navbar-collapse" id="bs-example-navbar-collapse-1">
      <ul class="nav navbar-nav">

	
        <li><a href="../conduct/">Code of Conduct</a></li>

	
        
        <li><a href="../setup/">Setup</a></li>
        <li class="dropdown">
          <a href="../" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-haspopup="true" aria-expanded="false">Episodes <span class="caret"></span></a>
          <ul class="dropdown-menu">
            
            <li><a href="../02-input-file-format/">Input File Format</a></li>
            
            <li><a href="../03-calc-genoprob/">Calculating Genotype Probabilities</a></li>
            
            <li><a href="../04-special-x-covar/">Special covariates for the X chromosome</a></li>
            
            <li><a href="../05-perform-genome-scan/">Performing a genome scan</a></li>
            
            <li><a href="../06-perform-perm-test/">Performing a permutation test</a></li>
            
            <li><a href="../07-find-lod-peaks/">Finding LOD peaks</a></li>
            
            <li><a href="../08-calc-kinship/">Calculating A Kinship Matrix</a></li>
            
            <li><a href="../09-perform-genome-scan-lmm/">Performing a genome scan with a linear mixed model</a></li>
            
            <li><a href="../10-perform-genome-scan-bin/">Performing a genome scan with binary traits</a></li>
            
            <li><a href="../11-est-qtl-effects/">Estimated QTL effects</a></li>
            
            <li><a href="../12-snp-assoc/">SNP association mapping</a></li>
            
            <li><a href="../13-qtl-in-do/">QTL analysis in Diversity Outbred Mice</a></li>
            
	    <li role="separator" class="divider"></li>
            <li><a href="../aio/">All in one page (Beta)</a></li>
          </ul>
        </li>
	

	
	
        <li class="dropdown">
          <a href="../" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-haspopup="true" aria-expanded="false">Extras <span class="caret"></span></a>
          <ul class="dropdown-menu">
            <li><a href="../reference/">Reference</a></li>
            
            <li><a href="../about/">About</a></li>
            
            <li><a href="../discuss/">Discussion</a></li>
            
            <li><a href="../figures/">Figures</a></li>
            
            <li><a href="../guide/">Instructor Notes</a></li>
            
          </ul>
        </li>
	

	
        <li><a href="../license/">License</a></li>
	
	
	<li><a href="/edit/gh-pages/_episodes_rmd/11-est-qtl-effects.Rmd">Improve this page <span class="glyphicon glyphicon-pencil" aria-hidden="true"></span></a></li>
	
	
      </ul>
      <form class="navbar-form navbar-right" role="search" id="search" onsubmit="google_search(); return false;">
        <div class="form-group">
          <input type="text" id="google-search" placeholder="Search..." aria-label="Google site search">
        </div>
      </form>
    </div>
  </div>
</nav>


<div class="row">
  <div class="col-md-1">
    <h3>
      
      <a href="../10-perform-genome-scan-bin/"><span class="glyphicon glyphicon-menu-left" aria-hidden="true"></span><span class="sr-only">previous episode</span></a>
      
    </h3>
  </div>
  <div class="col-md-10">
    
    <h3 class="maintitle"><a href="../">Quantitative Trait Mapping</a></h3>
    
  </div>
  <div class="col-md-1">
    <h3>
      
      <a href="../12-snp-assoc/"><span class="glyphicon glyphicon-menu-right" aria-hidden="true"></span><span class="sr-only">next episode</span></a>
      
    </h3>
  </div>
</div>

<article>
<div class="row">
  <div class="col-md-1">
  </div>
  <div class="col-md-10">
    <h1 class="maintitle">Estimated QTL effects</h1>
  </div>
  <div class="col-md-1">
  </div>
</div>


<blockquote class="objectives">
  <h2>Overview</h2>

  <div class="row">
    <div class="col-md-3">
      <strong>Teaching:</strong> 10 min
      <br/>
      <strong>Exercises:</strong> 20 min
    </div>
    <div class="col-md-9">
      <strong>Questions</strong>
      <ul>
	
	<li><p>How do I find the estimated effects of a QTL on a phenotype?</p>
</li>
	
      </ul>
    </div>
  </div>

  <div class="row">
    <div class="col-md-3">
    </div>
    <div class="col-md-9">
      <strong>Objectives</strong>
      <ul>
	
	<li><p>Obtain estimated QTL effects.</p>
</li>
	
	<li><p>Plot estimated QTL effects.</p>
</li>
	
      </ul>
    </div>
  </div>

</blockquote>

<p>Recall that to model data from a cross, we use</p>

<p><img src="../fig/linear-genetic-model.png" alt="" /></p>

<p>where <i>y<sub>j</sub></i> is the phenotype of the <i>j</i>th individual, μ 
is the mean phenotype value, <i>β<sub>k</sub></i> is the effect of the 
<i>kth</i> genotype, <i>X<sub>jk</sub></i> is the genotype for individual j, and ε<sub>j</sub> is the error for the <i>j</i>th individual. In the figure 
below, μ equals 94.6, and β equals 
15.4 for the alternative hypothesis 
(QTL exists).</p>

<p><img src="../fig/nullvalt.png" alt="" /></p>

<p>This linear model is <i>y</i> = 94.6 + 
15.4X + ε. The model intersects the 
genotype groups at their group means, and is based on μ and 
<i>β<sub>k</sub></i> for chromosome 2 marker D2Mit17 located at 56.8 cM.</p>

<p>The effect of genotype BB (the β for the 
BB genotype) at marker D2Mit17 is 
15.5, while the effect of the 
SS genotype is -15.4 on 
the liver phenotype. The effect of the SB genotype is 
-0.1 relative to μ equals 
94.6.</p>

<p>The <code class="language-plaintext highlighter-rouge">scan1()</code> function returns only LOD scores. To obtain estimated QTL effects,
use the function <code class="language-plaintext highlighter-rouge">scan1coef()</code>. This function takes a single phenotype and the 
genotype probabilities for a single chromosome and returns a matrix with the 
estimated coefficients at each putative QTL location along the chromosome.</p>

<p>For example, to get the estimated QTL effects on chromosome 2 for the liver 
phenotype, we would provide the chromosome 2 genotype probabilities and the 
liver phenotype to the function <code class="language-plaintext highlighter-rouge">scan1coef()</code> as follows:</p>

<div class="language-plaintext r highlighter-rouge"><div class="highlight"><pre class="highlight"><code>c2eff &lt;- scan1coef(pr[,"2"], iron$pheno[,"liver"])
</code></pre></div></div>

<p>The result is a matrix of 39 positions × 3 
genotypes. An additional column contains the intercept values (μ).</p>

<div class="language-plaintext r highlighter-rouge"><div class="highlight"><pre class="highlight"><code>dim(c2eff)
</code></pre></div></div>

<div class="language-plaintext output highlighter-rouge"><div class="highlight"><pre class="highlight"><code>[1] 39  4
</code></pre></div></div>

<div class="language-plaintext r highlighter-rouge"><div class="highlight"><pre class="highlight"><code>head(c2eff)
</code></pre></div></div>

<div class="language-plaintext output highlighter-rouge"><div class="highlight"><pre class="highlight"><code>                SS        SB        BB intercept
D2Mit379 -8.627249 -1.239968  9.867217  95.10455
c2.loc39 -8.802858 -1.405183 10.208040  95.15023
c2.loc40 -8.926486 -1.552789 10.479275  95.18752
c2.loc41 -8.994171 -1.670822 10.664993  95.21294
c2.loc42 -9.004121 -1.748738 10.752859  95.22372
c2.loc43 -8.956705 -1.779555 10.736260  95.21841
</code></pre></div></div>

<p>To plot the QTL effects, use the function <code class="language-plaintext highlighter-rouge">plot_coef()</code>.</p>

<div class="language-plaintext r highlighter-rouge"><div class="highlight"><pre class="highlight"><code>plot_coef(c2eff, map, legend = "topright")
</code></pre></div></div>

<div class="figure" style="text-align: center">
<img src="../fig/rmd-11-plot_effects_liver_simple-1.png" alt="plot of chunk plot_effects_liver_simple" />
<p class="caption">plot of chunk plot_effects_liver_simple</p>
</div>

<p>The plot shows effect values on the y-axis and cM values on the x-axis. The
value of the intercept (μ) appears at the top. The effect of the SB genotype is centered around zero, with the effects
of the other two genotypes above and below.</p>

<p>To plot only the effects, use the argument <code class="language-plaintext highlighter-rouge">columns</code> to indicate which 
coefficients to plot. Add <code class="language-plaintext highlighter-rouge">scan1_output</code> to include a LOD plot at the bottom.</p>

<div class="language-plaintext r highlighter-rouge"><div class="highlight"><pre class="highlight"><code>plot_coef(c2eff, map, columns = 1:3, scan1_output = out, 
          main = "Chromosome 2 QTL effects and LOD scores",
          legend = "topright")
</code></pre></div></div>

<div class="figure" style="text-align: center">
<img src="../fig/rmd-11-plot_effects_liver_c2-1.png" alt="plot of chunk plot_effects_liver_c2" />
<p class="caption">plot of chunk plot_effects_liver_c2</p>
</div>

<p>If instead you want additive and dominance effects, you can provide a square matrix of <em>contrasts</em>, as follows:</p>

<div class="language-plaintext r highlighter-rouge"><div class="highlight"><pre class="highlight"><code>c2effB &lt;- scan1coef(pr[,"2"], iron$pheno[,"liver"],
                    contrasts=cbind(mu=c(1,1,1), a=c(-1, 0, 1), d=c(-0.5, 1, -0.5)))
</code></pre></div></div>

<p>The result will then contain the estimates of <code class="language-plaintext highlighter-rouge">mu</code>, <code class="language-plaintext highlighter-rouge">a</code> (the additive effect), and <code class="language-plaintext highlighter-rouge">d</code> (the dominance effect).</p>

<div class="language-plaintext r highlighter-rouge"><div class="highlight"><pre class="highlight"><code>dim(c2effB)
</code></pre></div></div>

<div class="language-plaintext output highlighter-rouge"><div class="highlight"><pre class="highlight"><code>[1] 39  3
</code></pre></div></div>

<div class="language-plaintext r highlighter-rouge"><div class="highlight"><pre class="highlight"><code>head(c2effB)
</code></pre></div></div>

<div class="language-plaintext output highlighter-rouge"><div class="highlight"><pre class="highlight"><code>               mu        a         d
D2Mit379 95.10455 9.247233 -1.239968
c2.loc39 95.15023 9.505449 -1.405183
c2.loc40 95.18752 9.702880 -1.552789
c2.loc41 95.21294 9.829582 -1.670822
c2.loc42 95.22372 9.878490 -1.748738
c2.loc43 95.21841 9.846482 -1.779555
</code></pre></div></div>
<p>For marker D2Mit17, <code class="language-plaintext highlighter-rouge">mu</code>, <code class="language-plaintext highlighter-rouge">a</code>, and <code class="language-plaintext highlighter-rouge">d</code> are 94.616915, 15.4026314, -0.1033269.</p>

<p>Here’s a plot of the chromosome 2 additive and dominance effects, which are in the second and third columns.</p>

<div class="language-plaintext r highlighter-rouge"><div class="highlight"><pre class="highlight"><code>plot_coef(c2effB, map["2"], columns=2:3, col=col)
</code></pre></div></div>

<p><img src="../fig/chr2_effects_contrasts.png" alt="" /></p>

<p>If you provide a kinship matrix to <code class="language-plaintext highlighter-rouge">scan1coef()</code>, it fits a linear mixed model (LMM) to account for a residual polygenic effect. Here let’s use the kinship matrix from the LOCO method.</p>

<div class="language-plaintext r highlighter-rouge"><div class="highlight"><pre class="highlight"><code>c2eff_pg &lt;- scan1coef(pr[,"2"], iron$pheno[,"liver"], kinship_loco[["2"]])
dim(c2eff_pg)
</code></pre></div></div>

<div class="language-plaintext output highlighter-rouge"><div class="highlight"><pre class="highlight"><code>[1] 39  4
</code></pre></div></div>

<div class="language-plaintext r highlighter-rouge"><div class="highlight"><pre class="highlight"><code>head(c2eff_pg)
</code></pre></div></div>

<div class="language-plaintext output highlighter-rouge"><div class="highlight"><pre class="highlight"><code>                 SS         SB       BB intercept
D2Mit379  -9.552160 -0.5954469 10.14761  94.54643
c2.loc39  -9.868034 -0.7068007 10.57483  94.57886
c2.loc40 -10.130889 -0.8028381 10.93373  94.60481
c2.loc41 -10.332864 -0.8737574 11.20662  94.62137
c2.loc42 -10.468250 -0.9112467 11.37950  94.62626
c2.loc43 -10.533818 -0.9103487 11.44417  94.61832
</code></pre></div></div>

<p>Here’s a plot of the estimates.</p>

<div class="language-plaintext r highlighter-rouge"><div class="highlight"><pre class="highlight"><code>plot_coef(c2eff_pg, map, columns = 1:3, scan1_output = out_pg_loco, main = "Chromosome 2 QTL effects and LOD scores", legend = "topright")
</code></pre></div></div>

<div class="figure" style="text-align: center">
<img src="../fig/rmd-11-plot_effects_pg_liver_c2-1.png" alt="plot of chunk plot_effects_pg_liver_c2" />
<p class="caption">plot of chunk plot_effects_pg_liver_c2</p>
</div>

<p>You can also get estimated additive and dominance effects, using a matrix of contrasts.</p>

<div class="language-plaintext r highlighter-rouge"><div class="highlight"><pre class="highlight"><code>c2effB_pg &lt;- scan1coef(pr[,"2"], iron$pheno[,"liver"], kinship_loco[["2"]],
                       contrasts=cbind(mu=c(1,1,1), a=c(-1, 0, 1), d=c(-0.5, 1, -0.5)))
</code></pre></div></div>

<p>Here’s a plot of the results.</p>

<div class="language-plaintext r highlighter-rouge"><div class="highlight"><pre class="highlight"><code>plot(c2effB_pg, map["2"], columns=2:3, col=col)
</code></pre></div></div>

<p><img src="../fig/chr2_effects_pg_add_dom.png" alt="" /></p>

<p>Another option for estimating the QTL effects is to treat them as <a href="https://stats.stackexchange.com/questions/4700/what-is-the-difference-between-fixed-effect-random-effect-and-mixed-effect-mode#151800">random effects</a> and calculate <a href="https://en.wikipedia.org/wiki/Best_linear_unbiased_prediction">Best Linear Unbiased Predictors</a> (BLUPs). This is particularly valuable for multi-parent populations such as the Collaborative Cross and Diversity Outbred mice, where the large number of possible genotypes at a QTL leads to considerable variability in the effect estimates. To calculate BLUPs, use <code class="language-plaintext highlighter-rouge">scan1blup()</code>; it takes the same arguments as <code class="language-plaintext highlighter-rouge">scan1coef()</code>, including
the option of a kinship matrix to account for a residual polygenic effect.</p>

<div class="language-plaintext r highlighter-rouge"><div class="highlight"><pre class="highlight"><code>c2blup &lt;- scan1blup(pr[,"2"], iron$pheno[,"liver"], kinship_loco[["2"]])
</code></pre></div></div>

<p>Here is a plot of the BLUPs (as dashed curves) alongside the standard estimates.</p>

<div class="language-plaintext r highlighter-rouge"><div class="highlight"><pre class="highlight"><code>plot_coef(c2eff, map["2"], columns=1:3)
plot(c2blup, map["2"], columns=1:3, add=TRUE, lty=2, legend = "topright")
</code></pre></div></div>

<div class="figure" style="text-align: center">
<img src="../fig/rmd-11-plotblup-1.png" alt="plot of chunk plotblup" />
<p class="caption">plot of chunk plotblup</p>
</div>

<p>The <code class="language-plaintext highlighter-rouge">scan1coef</code> function can also provide estimated QTL effects for binary traits, with <code class="language-plaintext highlighter-rouge">model="binary"</code>. (However, <code class="language-plaintext highlighter-rouge">scan1blup</code> has not yet been implemented for binary traits.)</p>

<div class="language-plaintext r highlighter-rouge"><div class="highlight"><pre class="highlight"><code>c2eff_bin &lt;- scan1coef(pr[,"2"], bin_pheno[,"liver"], model="binary")
</code></pre></div></div>

<p>Here’s a plot of the effects. They’re a bit tricky to interpret, as they are basically log odds ratios.</p>

<p><img src="../fig/chr2_effects_binary.png" alt="" /></p>

<p>Finally, to plot the raw phenotypes against the genotypes at a single putative QTL position, you can use the function <code class="language-plaintext highlighter-rouge">plot_pxg()</code>. This takes a vector of genotypes as produced by the <code class="language-plaintext highlighter-rouge">maxmarg()</code> function, which picks the most likely genotype from a set of genotype probabilities, provided it is greater than some specified value (the argument <code class="language-plaintext highlighter-rouge">minprob</code>). Note that the “marg” in “maxmarg” stands for “marginal”, as this function is selecting the genotype at each position that has maximum marginal probability.</p>

<p>For example, we could get inferred genotypes at the chr 2 QTL for the liver phenotype (at 28.6 cM) as follows:</p>

<div class="language-plaintext r highlighter-rouge"><div class="highlight"><pre class="highlight"><code>g &lt;- maxmarg(pr, map, chr=2, pos=28.6, return_char=TRUE)
</code></pre></div></div>

<p>We use <code class="language-plaintext highlighter-rouge">return_char=TRUE</code> to have <code class="language-plaintext highlighter-rouge">maxmarg()</code> return a vector of character strings with the genotype labels.</p>

<p>We then plot the liver phenotype against these genotypes as follows:</p>

<div class="language-plaintext r highlighter-rouge"><div class="highlight"><pre class="highlight"><code>par(mar=c(4.1, 4.1, 0.6, 0.6))
plot_pxg(g, iron$pheno[,"liver"], ylab="Liver phenotype")
</code></pre></div></div>

<div class="figure" style="text-align: center">
<img src="../fig/rmd-11-plot_pheno_geno-1.png" alt="plot of chunk plot_pheno_geno" />
<p class="caption">plot of chunk plot_pheno_geno</p>
</div>

<div class="language-plaintext r highlighter-rouge"><div class="highlight"><pre class="highlight"><code>par(mar=c(4.1, 4.1, 0.6, 0.6))
plot_pxg(g, iron$pheno[,"liver"], SEmult=2, swap_axes=TRUE, xlab="Liver phenotype")
</code></pre></div></div>

<div class="figure" style="text-align: center">
<img src="../fig/rmd-11-plot_pheno_geno_se-1.png" alt="plot of chunk plot_pheno_geno_se" />
<p class="caption">plot of chunk plot_pheno_geno_se</p>
</div>

<blockquote class="challenge">
  <h2 id="challenge-1">Challenge 1</h2>
  <p>Find the QTL effects for chromosome 16 for the liver phenotype.<br />
1) Create an object <code class="language-plaintext highlighter-rouge">c16eff</code> to contain the effects.<br />
2) Plot the chromosome 16 coefficients and add the LOD plot at bottom.</p>

  <blockquote class="solution">

    <h2 id="solution-to-challenge-1">Solution to Challenge 1</h2>
    <p>1) <code class="language-plaintext highlighter-rouge">c16eff &lt;- scan1coef(pr[,"16"], iron$pheno[,"liver"])</code><br />
2) <code class="language-plaintext highlighter-rouge">plot_coef(c16eff, map, legend = "topright", scan1_output = out)</code></p>
  </blockquote>
</blockquote>

<blockquote class="challenge">
  <h2 id="challenge-2">Challenge 2</h2>
  <p>In the code block above, we use <code class="language-plaintext highlighter-rouge">swap_axes=TRUE</code> to place the phenotype
on the x-axis. 
We can use <code class="language-plaintext highlighter-rouge">SEmult=2</code> to include the mean ± 2 SE intervals.<br />
1) How would you decide which chromosome to plot? Discuss this with 
your neighbor, then write your responses in the collaborative document.<br />
2) Calculate and plot the best linear unbiased predictors (blups) for
spleen on the chromosome of your choice.</p>

  <blockquote class="solution">

    <h2 id="solution-to-challenge-2">Solution to Challenge 2</h2>

  </blockquote>
</blockquote>


<blockquote class="keypoints">
  <h2>Key Points</h2>
  <ul>
    
    <li><p>Estimated founder allele effects can be plotted from the mapping model coefficients.</p>
</li>
    
    <li><p>Additive and dominance effects can be plotted using contrasts.</p>
</li>
    
  </ul>
</blockquote>

</article>

<div class="row">
  <div class="col-md-1">
    <h3>
      
      <a href="../10-perform-genome-scan-bin/"><span class="glyphicon glyphicon-menu-left" aria-hidden="true"></span><span class="sr-only">previous episode</span></a>
      
    </h3>
  </div>
  <div class="col-md-10">
    
  </div>
  <div class="col-md-1">
    <h3>
      
      <a href="../12-snp-assoc/"><span class="glyphicon glyphicon-menu-right" aria-hidden="true"></span><span class="sr-only">next episode</span></a>
      
    </h3>
  </div>
</div>


      
      
<footer>
  <div class="row">
    <div class="col-md-6" align="left">
      <h4>
	Copyright &copy; 2016–2025
	
      </h4>
    </div>
    <div class="col-md-6" align="right">
      <h4>
	
	
	<a href="/edit/gh-pages/_episodes_rmd/11-est-qtl-effects.Rmd">Edit on GitHub</a>
	
	
	/
	<a href="/blob/gh-pages/CONTRIBUTING.md">Contributing</a>
	/
	<a href="/">Source</a>
	/
	<a href="/blob/gh-pages/CITATION">Cite</a>
	/
	<a href="mailto:susan.mcclatchy@jax.org">Contact</a>
      </h4>
    </div>
  </div>
</footer>

      
    </div>
    
<script src="../assets/js/jquery.min.js"></script>
<script src="../assets/js/bootstrap.min.js"></script>
<script src="../assets/js/lesson.js"></script>
<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
  ga('create', 'UA-37305346-2', 'auto');
  ga('send', 'pageview');
</script>

  </body>
</html>
