<!doctype html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta http-equiv="last-modified" content="2025-11-22 10:15:12 +0000">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <!-- meta "search-domain" used for google site search function google_search() -->
    <meta name="search-domain" value="">
    <link rel="stylesheet" type="text/css" href="../assets/css/bootstrap.css" />
    <link rel="stylesheet" type="text/css" href="../assets/css/bootstrap-theme.css" />
    <link rel="stylesheet" type="text/css" href="../assets/css/lesson.css" />
    
    <!-- HTML5 shim and Respond.js for IE8 support of HTML5 elements and media queries -->
    <!-- WARNING: Respond.js doesn't work if you view the page via file:// -->
    <!--[if lt IE 9]>
	<script src="https://oss.maxcdn.com/html5shiv/3.7.2/html5shiv.min.js"></script>
	<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
	<![endif]-->
    <title>Quantitative Trait Mapping: Performing a genome scan</title>
  </head>
  <body>
    <div class="container">
      
<nav class="navbar navbar-default">
  <div class="container-fluid">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#bs-example-navbar-collapse-1" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>

      
      

      
      <a class="navbar-brand" href="../">Home</a>

    </div>
    <div class="collapse navbar-collapse" id="bs-example-navbar-collapse-1">
      <ul class="nav navbar-nav">

	
        <li><a href="../conduct/">Code of Conduct</a></li>

	
        
        <li><a href="../setup/">Setup</a></li>
        <li class="dropdown">
          <a href="../" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-haspopup="true" aria-expanded="false">Episodes <span class="caret"></span></a>
          <ul class="dropdown-menu">
            
            <li><a href="../02-input-file-format/">Input File Format</a></li>
            
            <li><a href="../03-calc-genoprob/">Calculating Genotype Probabilities</a></li>
            
            <li><a href="../04-special-x-covar/">Special covariates for the X chromosome</a></li>
            
            <li><a href="../05-perform-genome-scan/">Performing a genome scan</a></li>
            
            <li><a href="../06-perform-perm-test/">Performing a permutation test</a></li>
            
            <li><a href="../07-find-lod-peaks/">Finding LOD peaks</a></li>
            
            <li><a href="../08-calc-kinship/">Calculating A Kinship Matrix</a></li>
            
            <li><a href="../09-perform-genome-scan-lmm/">Performing a genome scan with a linear mixed model</a></li>
            
            <li><a href="../10-perform-genome-scan-bin/">Performing a genome scan with binary traits</a></li>
            
            <li><a href="../11-est-qtl-effects/">Estimated QTL effects</a></li>
            
            <li><a href="../12-snp-assoc/">SNP association mapping</a></li>
            
            <li><a href="../13-qtl-in-do/">QTL analysis in Diversity Outbred Mice</a></li>
            
	    <li role="separator" class="divider"></li>
            <li><a href="../aio/">All in one page (Beta)</a></li>
          </ul>
        </li>
	

	
	
        <li class="dropdown">
          <a href="../" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-haspopup="true" aria-expanded="false">Extras <span class="caret"></span></a>
          <ul class="dropdown-menu">
            <li><a href="../reference/">Reference</a></li>
            
            <li><a href="../about/">About</a></li>
            
            <li><a href="../discuss/">Discussion</a></li>
            
            <li><a href="../figures/">Figures</a></li>
            
            <li><a href="../guide/">Instructor Notes</a></li>
            
          </ul>
        </li>
	

	
        <li><a href="../license/">License</a></li>
	
	
	<li><a href="/edit/gh-pages/_episodes_rmd/05-perform-genome-scan.Rmd">Improve this page <span class="glyphicon glyphicon-pencil" aria-hidden="true"></span></a></li>
	
	
      </ul>
      <form class="navbar-form navbar-right" role="search" id="search" onsubmit="google_search(); return false;">
        <div class="form-group">
          <input type="text" id="google-search" placeholder="Search..." aria-label="Google site search">
        </div>
      </form>
    </div>
  </div>
</nav>


<div class="row">
  <div class="col-md-1">
    <h3>
      
      <a href="../04-special-x-covar/"><span class="glyphicon glyphicon-menu-left" aria-hidden="true"></span><span class="sr-only">previous episode</span></a>
      
    </h3>
  </div>
  <div class="col-md-10">
    
    <h3 class="maintitle"><a href="../">Quantitative Trait Mapping</a></h3>
    
  </div>
  <div class="col-md-1">
    <h3>
      
      <a href="../06-perform-perm-test/"><span class="glyphicon glyphicon-menu-right" aria-hidden="true"></span><span class="sr-only">next episode</span></a>
      
    </h3>
  </div>
</div>

<article>
<div class="row">
  <div class="col-md-1">
  </div>
  <div class="col-md-10">
    <h1 class="maintitle">Performing a genome scan</h1>
  </div>
  <div class="col-md-1">
  </div>
</div>


<blockquote class="objectives">
  <h2>Overview</h2>

  <div class="row">
    <div class="col-md-3">
      <strong>Teaching:</strong> 30 min
      <br/>
      <strong>Exercises:</strong> 30 min
    </div>
    <div class="col-md-9">
      <strong>Questions</strong>
      <ul>
	
	<li><p>How do I perform a genome scan?</p>
</li>
	
	<li><p>How do I plot a genome scan?</p>
</li>
	
      </ul>
    </div>
  </div>

  <div class="row">
    <div class="col-md-3">
    </div>
    <div class="col-md-9">
      <strong>Objectives</strong>
      <ul>
	
	<li><p>Create a genome scan for a set of traits.</p>
</li>
	
	<li><p>Plot a genome scan.</p>
</li>
	
      </ul>
    </div>
  </div>

</blockquote>

<p>The freely available chapter on <a href="http://www.rqtl.org/book/rqtlbook_ch04.pdf">single-QTL analysis</a> from Broman and Sen’s <a href="http://www.rqtl.org/book/">A Guide to QTL Mapping with R/qtl</a> describes different methods for QTL analysis. Two of these methods are described here using data from an experiment on hypertension in the mouse <a href="https://s3.amazonaws.com/academia.edu.documents/45963759/geno.2000.640120160526-29022-36mpgg.pdf?AWSAccessKeyId=AKIAIWOWYYGZ2Y53UL3A&amp;Expires=1513786158&amp;Signature=rtodlYwe0LDmYZFOm1ejvZjZhQ0%3D&amp;response-content-disposition=inline%3B%20filename%3DConcordance_of_murine_quantitative_trait.pdf">Sugiyama et al., Genomics 71:70-77, 2001</a>. The study employed a backcross between two mouse strains resulting in two possible genotypes - AA for homozygotes, and AB for heterozygotes.</p>

<p>Linear regression can be employed to identify presence of QTL in a cross. To identify QTL using regression, we compare the fit for two models: 1) the null hypothesis that there are no QTL anywhere in the genome; and 2) the alternative hypothesis that there is a QTL near a specific position. A sloped line indicates that there is a difference in mean phenotype between the two genotype groups, and that a QTL is present. A line with no slope indicates that there is no difference in mean phenotype between the two groups, and that no QTL exists. Regression aims to find the line of best fit to the data. In the case of a backcross with only two genotypes, a t-test is performed at the marker to determine whether the difference in phenotype means is zero.</p>

<p><img src="../fig/nullvalt.png" alt="adapted from Broman &amp; Sen, 2009" /></p>

<p>To find the line of best fit, the residuals or errors are calculated, then squared for each data point.</p>

<p><img src="../fig/residual.png" alt="" />
<img src="../fig/squared-residual.png" alt="" />
<img src="../fig/null.png" alt="" /></p>

<p>The line of best fit will be the one that minimizes the sum of squared residuals, which maximizes the likelihood of the data.</p>

<p>Marker regression produces a LOD (logarithm of odds) score comparing the null hypothesis to the alternative. The LOD score is calculated using the sum of squared residuals for the null and alternative hypotheses. The LOD score is the difference between the log10 likelihood of the null hypothesis and the log10 likelihood of the alternative hypothesis. It is related to the regression model above by identifying the line of best fit to the data. A higher LOD score indicates greater likelihood of the alternative hypothesis. A LOD score closer to zero favors the null hypothesis.</p>

<p>Marker regression can identify the existence and effect of a QTL by comparing means between groups, however, it requires known marker genotypes and can’t identify QTL in between typed markers. To identify QTL between typed markers, we use Haley-Knott regression. After <a href="https://smcclatchy.github.io/mapping/03-calc-genoprob/">calculating genotype probabilities</a>, we can regress the phenotypes for animals of unknown genotype on these conditional genotype probabilities (conditional on known marker genotypes). In Haley-Knott regression, phenotype values can be plotted and a regression line drawn through the phenotype mean for the untyped individuals.</p>

<p><img src="../fig/hk-regress.png" alt="" /></p>

<p>As shown by the green circle in the figure, an individual of unknown genotype is placed between known genotypes according to the probability of its genotype being AA or AB. In this case, the probability of this individual having genotype AA is 0.6, and the probability of having genotype AB is 0.4.</p>

<p>To perform a genome scan by Haley-Knott regression
(<a href="https://www.ncbi.nlm.nih.gov/pubmed/16718932">Haley and Knott 1992</a>),
use the function <code class="language-plaintext highlighter-rouge">scan1()</code>.  <code class="language-plaintext highlighter-rouge">scan1()</code> takes as input the genotype probabilities, a matrix of phenotypes, and then optional additive and interactive covariates, and the special X chromosome covariates. Another option is to provide a vector of weights.</p>

<div class="language-plaintext r highlighter-rouge"><div class="highlight"><pre class="highlight"><code>out &lt;- scan1(genoprobs = pr, pheno = iron$pheno, Xcovar=Xcovar)
</code></pre></div></div>

<p>On a multi-core machine, you can get some speed-up via the <code class="language-plaintext highlighter-rouge">cores</code> argument, as with <code class="language-plaintext highlighter-rouge">calc_genoprob()</code> and <code class="language-plaintext highlighter-rouge">calc_kinship()</code>.</p>

<div class="language-plaintext r highlighter-rouge"><div class="highlight"><pre class="highlight"><code>out &lt;- scan1(genoprobs = pr, pheno = iron$pheno, Xcovar=Xcovar, cores=4)
</code></pre></div></div>

<p>The output of <code class="language-plaintext highlighter-rouge">scan1()</code> is a matrix of LOD scores, positions × phenotypes.</p>

<p>Take a look at the first ten rows of the scan object. The numerical values are the LOD scores for the marker or pseudomarker named at the beginning of the row. LOD values are given for liver and spleen.</p>

<div class="language-plaintext r highlighter-rouge"><div class="highlight"><pre class="highlight"><code>head(out, n=10)
</code></pre></div></div>

<div class="language-plaintext output highlighter-rouge"><div class="highlight"><pre class="highlight"><code>             liver    spleen
D1Mit18  0.2928511 0.4566189
c1.loc28 0.2815294 0.4378754
c1.loc29 0.2696594 0.4193363
c1.loc30 0.2572780 0.4011646
c1.loc31 0.2444340 0.3835257
c1.loc32 0.2311884 0.3665798
c1.loc33 0.2176149 0.3504744
c1.loc34 0.2037988 0.3353358
c1.loc35 0.1898359 0.3212614
c1.loc36 0.1758305 0.3083145
</code></pre></div></div>

<p>The function <code class="language-plaintext highlighter-rouge">plot_scan1()</code> can be used to plot the LOD curves. Use the argument <code class="language-plaintext highlighter-rouge">lodcolumn</code> to indicate which column to plot.</p>

<div class="language-plaintext r highlighter-rouge"><div class="highlight"><pre class="highlight"><code>plot_scan1(out, map = map, lodcolumn = "liver")
</code></pre></div></div>

<p><img src="../fig/lod-plot.png" alt="" /></p>

<p>The LOD plot for liver clearly shows the largest peak is on chromosome 16. There are smaller peaks on chromosomes 2, 7, and 8. Which of these peaks is significant, and why? We’ll evaluate the significance of genome scan results in a later episode on <a href="https://smcclatchy.github.io/mapping/10-perform-perm-test/">performing a permutation test</a>.</p>

<blockquote class="challenge">
  <h2 id="challenge-1">Challenge 1</h2>
  <p>Use the <code class="language-plaintext highlighter-rouge">head()</code> function to view the first 30 rows of the scan output. What is the next genotyped marker in the scan output? What are its LOD scores for liver and spleen?</p>

  <blockquote class="solution">
    <h2 id="solution-to-challenge-1">Solution to Challenge 1</h2>
    <p><code class="language-plaintext highlighter-rouge">head(out, n=30)</code><br />
D1Mit80  0.02154502 0.2216682</p>
  </blockquote>
</blockquote>

<blockquote class="challenge">
  <h2 id="challenge-2">Challenge 2</h2>
  <p>Use the <code class="language-plaintext highlighter-rouge">sort()</code> function to sort the LOD scores for liver. Hint: run <code class="language-plaintext highlighter-rouge">dim(out)</code> for the row and column dimensions, or <code class="language-plaintext highlighter-rouge">colnames(out)</code> for the column names. <br />
Which pseudomarker has the highest LOD score? Which genotyped marker has the highest LOD score? What chromosome number are they on?</p>

  <blockquote class="solution">
    <h2 id="solution-to-challenge-2">Solution to Challenge 2</h2>
    <p><code class="language-plaintext highlighter-rouge">sort(out[,1])</code> for column 1 or <code class="language-plaintext highlighter-rouge">sort(out[,"liver"])</code> for the column named “liver”.<br />
The pseudomarker with the largest score is c16.loc29, with a LOD of 7.68. The genotyped marker with the largest LOD is D16Mit30 with a score of 7.28. Both are located on chromosome 16.</p>
  </blockquote>
</blockquote>

<blockquote class="challenge">
  <h2 id="challenge-3">Challenge 3</h2>
  <p>Use the <code class="language-plaintext highlighter-rouge">sort()</code> function to sort the LOD scores for spleen. Which pseudomarker has the highest LOD score? Which genotyped marker has the highest LOD score? What chromosome number are they on?</p>

  <blockquote class="solution">
    <h2 id="solution-to-challenge-3">Solution to Challenge 3</h2>
    <p><code class="language-plaintext highlighter-rouge">sort(out[,2])</code> or <code class="language-plaintext highlighter-rouge">sort(out[,"spleen"])</code>
The pseudomarker with the largest score is c9.loc57, with a LOD of 12.1. The genotyped marker with the largest LOD is D9Mit182 with a score of 10.4. Both are located on chromosome 9.</p>
  </blockquote>
</blockquote>

<blockquote class="challenge">
  <h2 id="challenge-4">Challenge 4</h2>
  <p>Plot the LOD scores for spleen. Does the genome scan for spleen share any large-ish peaks with the scan for liver?</p>

  <blockquote class="solution">
    <h2 id="solution-to-challenge-4">Solution to Challenge 4</h2>
    <p><code class="language-plaintext highlighter-rouge">plot_scan1(out, map = map, lodcolumn = "spleen")</code>
Both liver and spleen genome scans share a peak on chromosome 8 with a LOD score near 4.</p>
  </blockquote>
</blockquote>


<blockquote class="keypoints">
  <h2>Key Points</h2>
  <ul>
    
    <li><p>A qtl2 genome scan requires genotype probabilities and a phenotype matrix.</p>
</li>
    
    <li><p>The output from a genome scan contains a LOD score matrix, map positions, and phenotypes.</p>
</li>
    
    <li><p>LOD curve plots for a genome scan can be viewed with plot_scan1().</p>
</li>
    
  </ul>
</blockquote>

</article>

<div class="row">
  <div class="col-md-1">
    <h3>
      
      <a href="../04-special-x-covar/"><span class="glyphicon glyphicon-menu-left" aria-hidden="true"></span><span class="sr-only">previous episode</span></a>
      
    </h3>
  </div>
  <div class="col-md-10">
    
  </div>
  <div class="col-md-1">
    <h3>
      
      <a href="../06-perform-perm-test/"><span class="glyphicon glyphicon-menu-right" aria-hidden="true"></span><span class="sr-only">next episode</span></a>
      
    </h3>
  </div>
</div>


      
      
<footer>
  <div class="row">
    <div class="col-md-6" align="left">
      <h4>
	Copyright &copy; 2016–2025
	
      </h4>
    </div>
    <div class="col-md-6" align="right">
      <h4>
	
	
	<a href="/edit/gh-pages/_episodes_rmd/05-perform-genome-scan.Rmd">Edit on GitHub</a>
	
	
	/
	<a href="/blob/gh-pages/CONTRIBUTING.md">Contributing</a>
	/
	<a href="/">Source</a>
	/
	<a href="/blob/gh-pages/CITATION">Cite</a>
	/
	<a href="mailto:susan.mcclatchy@jax.org">Contact</a>
      </h4>
    </div>
  </div>
</footer>

      
    </div>
    
<script src="../assets/js/jquery.min.js"></script>
<script src="../assets/js/bootstrap.min.js"></script>
<script src="../assets/js/lesson.js"></script>
<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
  ga('create', 'UA-37305346-2', 'auto');
  ga('send', 'pageview');
</script>

  </body>
</html>
