<!doctype html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta http-equiv="last-modified" content="2025-11-22 10:15:12 +0000">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <!-- meta "search-domain" used for google site search function google_search() -->
    <meta name="search-domain" value="">
    <link rel="stylesheet" type="text/css" href="../assets/css/bootstrap.css" />
    <link rel="stylesheet" type="text/css" href="../assets/css/bootstrap-theme.css" />
    <link rel="stylesheet" type="text/css" href="../assets/css/lesson.css" />
    
    <!-- HTML5 shim and Respond.js for IE8 support of HTML5 elements and media queries -->
    <!-- WARNING: Respond.js doesn't work if you view the page via file:// -->
    <!--[if lt IE 9]>
	<script src="https://oss.maxcdn.com/html5shiv/3.7.2/html5shiv.min.js"></script>
	<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
	<![endif]-->
    <title>Quantitative Trait Mapping: Performing a permutation test</title>
  </head>
  <body>
    <div class="container">
      
<nav class="navbar navbar-default">
  <div class="container-fluid">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#bs-example-navbar-collapse-1" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>

      
      

      
      <a class="navbar-brand" href="../">Home</a>

    </div>
    <div class="collapse navbar-collapse" id="bs-example-navbar-collapse-1">
      <ul class="nav navbar-nav">

	
        <li><a href="../conduct/">Code of Conduct</a></li>

	
        
        <li><a href="../setup/">Setup</a></li>
        <li class="dropdown">
          <a href="../" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-haspopup="true" aria-expanded="false">Episodes <span class="caret"></span></a>
          <ul class="dropdown-menu">
            
            <li><a href="../02-input-file-format/">Input File Format</a></li>
            
            <li><a href="../03-calc-genoprob/">Calculating Genotype Probabilities</a></li>
            
            <li><a href="../04-special-x-covar/">Special covariates for the X chromosome</a></li>
            
            <li><a href="../05-perform-genome-scan/">Performing a genome scan</a></li>
            
            <li><a href="../06-perform-perm-test/">Performing a permutation test</a></li>
            
            <li><a href="../07-find-lod-peaks/">Finding LOD peaks</a></li>
            
            <li><a href="../08-calc-kinship/">Calculating A Kinship Matrix</a></li>
            
            <li><a href="../09-perform-genome-scan-lmm/">Performing a genome scan with a linear mixed model</a></li>
            
            <li><a href="../10-perform-genome-scan-bin/">Performing a genome scan with binary traits</a></li>
            
            <li><a href="../11-est-qtl-effects/">Estimated QTL effects</a></li>
            
            <li><a href="../12-snp-assoc/">SNP association mapping</a></li>
            
            <li><a href="../13-qtl-in-do/">QTL analysis in Diversity Outbred Mice</a></li>
            
	    <li role="separator" class="divider"></li>
            <li><a href="../aio/">All in one page (Beta)</a></li>
          </ul>
        </li>
	

	
	
        <li class="dropdown">
          <a href="../" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-haspopup="true" aria-expanded="false">Extras <span class="caret"></span></a>
          <ul class="dropdown-menu">
            <li><a href="../reference/">Reference</a></li>
            
            <li><a href="../about/">About</a></li>
            
            <li><a href="../discuss/">Discussion</a></li>
            
            <li><a href="../figures/">Figures</a></li>
            
            <li><a href="../guide/">Instructor Notes</a></li>
            
          </ul>
        </li>
	

	
        <li><a href="../license/">License</a></li>
	
	
	<li><a href="/edit/gh-pages/_episodes_rmd/06-perform-perm-test.Rmd">Improve this page <span class="glyphicon glyphicon-pencil" aria-hidden="true"></span></a></li>
	
	
      </ul>
      <form class="navbar-form navbar-right" role="search" id="search" onsubmit="google_search(); return false;">
        <div class="form-group">
          <input type="text" id="google-search" placeholder="Search..." aria-label="Google site search">
        </div>
      </form>
    </div>
  </div>
</nav>


<div class="row">
  <div class="col-md-1">
    <h3>
      
      <a href="../05-perform-genome-scan/"><span class="glyphicon glyphicon-menu-left" aria-hidden="true"></span><span class="sr-only">previous episode</span></a>
      
    </h3>
  </div>
  <div class="col-md-10">
    
    <h3 class="maintitle"><a href="../">Quantitative Trait Mapping</a></h3>
    
  </div>
  <div class="col-md-1">
    <h3>
      
      <a href="../07-find-lod-peaks/"><span class="glyphicon glyphicon-menu-right" aria-hidden="true"></span><span class="sr-only">next episode</span></a>
      
    </h3>
  </div>
</div>

<article>
<div class="row">
  <div class="col-md-1">
  </div>
  <div class="col-md-10">
    <h1 class="maintitle">Performing a permutation test</h1>
  </div>
  <div class="col-md-1">
  </div>
</div>


<blockquote class="objectives">
  <h2>Overview</h2>

  <div class="row">
    <div class="col-md-3">
      <strong>Teaching:</strong> 10 min
      <br/>
      <strong>Exercises:</strong> 20 min
    </div>
    <div class="col-md-9">
      <strong>Questions</strong>
      <ul>
	
	<li><p>How can I evaluate the statistical significance of genome scan results?</p>
</li>
	
      </ul>
    </div>
  </div>

  <div class="row">
    <div class="col-md-3">
    </div>
    <div class="col-md-9">
      <strong>Objectives</strong>
      <ul>
	
	<li><p>Run a permutation test to establish LOD score thresholds.</p>
</li>
	
      </ul>
    </div>
  </div>

</blockquote>

<p>To establish the statistical significance of the results of a genome scan, a permutation test can identify the maximum LOD score that can occur by random chance. A permutation tests shuffles genotypes and phenotypes, essentially breaking the relationship between the two. The genome-wide maximum LOD score is then calculated on the permuted data, and this score used as a threshold of statistical significance. A genome-wide maximum LOD on shuffled, or permuted, data serves as the threshold because it represents the highest LOD score generated by random chance.</p>

<p>The <code class="language-plaintext highlighter-rouge">scan1perm()</code> function takes the same arguments as <code class="language-plaintext highlighter-rouge">scan1()</code>, plus additional arguments to control the permutations:</p>

<ul>
  <li><code class="language-plaintext highlighter-rouge">n_perm</code> is the number of permutation replicates.</li>
  <li><code class="language-plaintext highlighter-rouge">perm_Xsp</code> controls whether to perform autosome/X chromosome specific permutations (with <code class="language-plaintext highlighter-rouge">perm_Xsp=TRUE</code>) or not (the default is FALSE).</li>
  <li><code class="language-plaintext highlighter-rouge">perm_strata</code> is a vector that defines the strata for a stratified permutation test.</li>
  <li><code class="language-plaintext highlighter-rouge">chr_lengths</code> is a vector of chromosome lengths, used in the case that <code class="language-plaintext highlighter-rouge">perm_Xsp=TRUE</code>.</li>
</ul>

<p>As with <code class="language-plaintext highlighter-rouge">scan1()</code>, you may provide a kinship matrix (or vector of kinship matrices, for the “leave one chromosome out” (loco) approach), in order to fit linear mixed models. If <code class="language-plaintext highlighter-rouge">kinship</code> is unspecified, the function performs ordinary Haley-Knott regression.</p>

<p>To perform a permutation test with the <code class="language-plaintext highlighter-rouge">iron</code> data, we run <code class="language-plaintext highlighter-rouge">scan1perm()</code>, provide it with the genotype probabilities, the phenotype data, X covariates and number of permutations. For expediency, we’ll use only 10 permutations, although 1000 is recommended.</p>

<div class="language-plaintext r highlighter-rouge"><div class="highlight"><pre class="highlight"><code>operm &lt;- scan1perm(genoprobs = pr, pheno = iron$pheno, Xcovar = Xcovar, n_perm = 1000) # replace 1000 with 10 for expediency
</code></pre></div></div>

<p>Note the need to specify special covariates for the X chromosome (via <code class="language-plaintext highlighter-rouge">Xcovar</code>), to be included under the null hypothesis of no QTL. And note that when these are provided, the default is to perform a stratified permutation test, using strata defined by the rows in
<code class="language-plaintext highlighter-rouge">Xcovar</code>. In general, when the X chromosome is considered, one will wish to stratify at least by sex.</p>

<p>Also note that, as with <code class="language-plaintext highlighter-rouge">scan1()</code>, you can speed up the calculations on a multi-core machine by specifying the argument <code class="language-plaintext highlighter-rouge">cores</code>. With <code class="language-plaintext highlighter-rouge">cores=0</code>, the number of available cores will be detected via <code class="language-plaintext highlighter-rouge">parallel::detectCores()</code>. Otherwise, specify the number of cores as a positive integer. For large datasets, be mindful of the amount of memory that will be needed; you may need to use fewer than the maximum number of cores, to avoid going beyond the available memory.</p>

<div class="language-plaintext r highlighter-rouge"><div class="highlight"><pre class="highlight"><code>operm &lt;- scan1perm(pr, iron$pheno, Xcovar=Xcovar, n_perm=1000, cores=0)
</code></pre></div></div>

<p><code class="language-plaintext highlighter-rouge">operm</code> now contains the maximum LOD score for each permutation for the liver and spleen phenotypes. There should be 1000 values for each phenotypes. We can view the liver permutation LOD scores by making a histogram.</p>

<div class="language-plaintext r highlighter-rouge"><div class="highlight"><pre class="highlight"><code>hist(operm[,'liver'], breaks = 50, xlab = "LOD", main = "LOD scores for liver scan with threshold in red")
abline(v = summary(operm)[,'liver'], col = 'red', lwd = 2)
</code></pre></div></div>

<div class="figure" style="text-align: center">
<img src="../fig/rmd-06-hist_perm-1.png" alt="plot of chunk hist_perm" />
<p class="caption">plot of chunk hist_perm</p>
</div>

<p>In the histogram above, you can see that most of the maximum LOD scores fall between 1 and 3. This means that we expect LOD scores less than 3 to occur by chance fairly often. The red line indicates the alpha = 0.05 threshold, which means that, under permutation, we only see LOD values as high or higher, 5% of the time. This is one way of estimating a significance threshold for QTL plots.</p>

<p>To get estimated significance thresholds, use the function <code class="language-plaintext highlighter-rouge">summary()</code>.</p>

<div class="language-plaintext r highlighter-rouge"><div class="highlight"><pre class="highlight"><code>summary(operm)
</code></pre></div></div>

<div class="language-plaintext output highlighter-rouge"><div class="highlight"><pre class="highlight"><code>LOD thresholds (1000 permutations)
     liver spleen
0.05  3.34   3.79
</code></pre></div></div>

<p>The default is to return the 5% significance thresholds. Thresholds for other (or for multiple) significance levels can be obtained via the <code class="language-plaintext highlighter-rouge">alpha</code> argument.</p>

<div class="language-plaintext r highlighter-rouge"><div class="highlight"><pre class="highlight"><code>summary(operm, alpha=c(0.2, 0.05))
</code></pre></div></div>

<div class="language-plaintext output highlighter-rouge"><div class="highlight"><pre class="highlight"><code>LOD thresholds (1000 permutations)
     liver spleen
0.2   2.65   2.73
0.05  3.34   3.79
</code></pre></div></div>

<p>To obtain autosome/X chromosome-specific significance thresholds, specify <code class="language-plaintext highlighter-rouge">perm_Xsp=TRUE</code>. In this case, you need to provide chromosome lengths, which may be obtained with the function <code class="language-plaintext highlighter-rouge">chr_lengths()</code>.</p>

<div class="language-plaintext r highlighter-rouge"><div class="highlight"><pre class="highlight"><code>operm2 &lt;- scan1perm(pr, iron$pheno, Xcovar=Xcovar, n_perm=1000,
                    perm_Xsp=TRUE, chr_lengths=chr_lengths(map))
</code></pre></div></div>

<p>Separate permutations are performed for the autosomes and X chromosome, and considerably more permutation replicates are needed for the X chromosome. The computations take about twice as much time.
See <a href="https://www.ncbi.nlm.nih.gov/pubmed/17028340">Broman et al. (2006) Genetics
174:2151-2158</a>.</p>

<p>The significance thresholds are again derived via <code class="language-plaintext highlighter-rouge">summary()</code>:</p>

<div class="language-plaintext r highlighter-rouge"><div class="highlight"><pre class="highlight"><code>summary(operm2, alpha=c(0.2, 0.05))
</code></pre></div></div>

<div class="language-plaintext output highlighter-rouge"><div class="highlight"><pre class="highlight"><code>Autosome LOD thresholds (1000 permutations)
     liver spleen
0.2   2.64   2.62
0.05  3.33   3.32

X chromosome LOD thresholds (28243 permutations)
     liver spleen
0.2   3.13   4.01
0.05  3.74   5.03
</code></pre></div></div>

<p>As with <code class="language-plaintext highlighter-rouge">scan1</code>, we can use <code class="language-plaintext highlighter-rouge">scan1perm</code> with binary traits, using the argument <code class="language-plaintext highlighter-rouge">model="binary"</code>. Again, this can’t be used with a kinship matrix, but all of the other arguments can be applied.</p>

<div class="language-plaintext r highlighter-rouge"><div class="highlight"><pre class="highlight"><code>operm_bin &lt;- scan1perm(pr, bin_pheno, Xcovar=Xcovar, n_perm=1000, 
                       perm_Xsp=TRUE, chr_lengths=chr_lengths(map),
                       model="binary")
</code></pre></div></div>

<p>Here are the estimated 5% and 20% significance thresholds.</p>

<div class="language-plaintext r highlighter-rouge"><div class="highlight"><pre class="highlight"><code>summary(operm_bin, alpha=c(0.2, 0.05))
</code></pre></div></div>

<div class="language-plaintext output highlighter-rouge"><div class="highlight"><pre class="highlight"><code>Autosome LOD thresholds (1000 permutations)
     liver spleen
0.2   2.68   2.61
0.05  3.45   3.33

X chromosome LOD thresholds (28243 permutations)
     liver spleen
0.2   3.09   3.12
0.05  3.80   3.81
</code></pre></div></div>

<p>The code below shuffles the phenotypes so that they no longer match up with the genotypes. The purpose of this is to find out how high the LOD score can be due to random chance alone.</p>

<div class="language-plaintext r highlighter-rouge"><div class="highlight"><pre class="highlight"><code>shuffled_order &lt;- sample(rownames(iron$pheno))
pheno_permuted &lt;- iron$pheno
rownames(pheno_permuted) &lt;- shuffled_order
xcovar_permuted &lt;- Xcovar
rownames(xcovar_permuted) &lt;- shuffled_order
out_permuted &lt;- scan1(genoprobs = pr, pheno = pheno_permuted, Xcovar = xcovar_permuted)
plot(out_permuted, map)
head(shuffled_order)
</code></pre></div></div>

<blockquote class="challenge">
  <h2 id="challenge-1">Challenge 1</h2>
  <p>Run the preceding code to shuffle the phenotype data and plot a genome
scan with this shuffled (permuted) data.</p>

  <p>What is the maximum LOD score in the scan from this permuted data?<br />
How does it compare to the maximum LOD scores obtained from the earlier scan?<br />
How does it compare to the 5% and 20% LOD thresholds obtained earlier?<br />
Paste the maximum LOD score in the scan from your permuted data into the etherpad.</p>

  <blockquote class="solution">
    <h2 id="solution-to-challenge-1">Solution to Challenge 1</h2>

  </blockquote>
</blockquote>

<blockquote class="challenge">
  <h2 id="challenge-2">Challenge 2</h2>
  <p>1) Find the 1% and 10% significance thresholds for the first set of 
permutations contained in the object <code class="language-plaintext highlighter-rouge">operm</code>.<br />
2) What do the 1% and 10% significance thresholds say about LOD scores?</p>

  <blockquote class="solution">
    <h2 id="solution-to-challenge-2">Solution to Challenge 2</h2>
    <p>1) <code class="language-plaintext highlighter-rouge">summary(operm, alpha=c(0.01, 0.10))</code><br />
2) These LOD thresholds indicate maximum LOD scores that can be obtained
by random chance at the 1% and 10% significance levels. We expect to see LOD 
values this high or higher 1% and 10% of the time respectively.</p>
  </blockquote>
</blockquote>


<blockquote class="keypoints">
  <h2>Key Points</h2>
  <ul>
    
    <li><p>A permutation test establishes the statistical  significance of a genome scan.</p>
</li>
    
  </ul>
</blockquote>

</article>

<div class="row">
  <div class="col-md-1">
    <h3>
      
      <a href="../05-perform-genome-scan/"><span class="glyphicon glyphicon-menu-left" aria-hidden="true"></span><span class="sr-only">previous episode</span></a>
      
    </h3>
  </div>
  <div class="col-md-10">
    
  </div>
  <div class="col-md-1">
    <h3>
      
      <a href="../07-find-lod-peaks/"><span class="glyphicon glyphicon-menu-right" aria-hidden="true"></span><span class="sr-only">next episode</span></a>
      
    </h3>
  </div>
</div>


      
      
<footer>
  <div class="row">
    <div class="col-md-6" align="left">
      <h4>
	Copyright &copy; 2016–2025
	
      </h4>
    </div>
    <div class="col-md-6" align="right">
      <h4>
	
	
	<a href="/edit/gh-pages/_episodes_rmd/06-perform-perm-test.Rmd">Edit on GitHub</a>
	
	
	/
	<a href="/blob/gh-pages/CONTRIBUTING.md">Contributing</a>
	/
	<a href="/">Source</a>
	/
	<a href="/blob/gh-pages/CITATION">Cite</a>
	/
	<a href="mailto:susan.mcclatchy@jax.org">Contact</a>
      </h4>
    </div>
  </div>
</footer>

      
    </div>
    
<script src="../assets/js/jquery.min.js"></script>
<script src="../assets/js/bootstrap.min.js"></script>
<script src="../assets/js/lesson.js"></script>
<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
  ga('create', 'UA-37305346-2', 'auto');
  ga('send', 'pageview');
</script>

  </body>
</html>
