<!doctype html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta http-equiv="last-modified" content="2025-11-22 10:15:12 +0000">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <!-- meta "search-domain" used for google site search function google_search() -->
    <meta name="search-domain" value="">
    <link rel="stylesheet" type="text/css" href="../assets/css/bootstrap.css" />
    <link rel="stylesheet" type="text/css" href="../assets/css/bootstrap-theme.css" />
    <link rel="stylesheet" type="text/css" href="../assets/css/lesson.css" />
    
    <!-- HTML5 shim and Respond.js for IE8 support of HTML5 elements and media queries -->
    <!-- WARNING: Respond.js doesn't work if you view the page via file:// -->
    <!--[if lt IE 9]>
	<script src="https://oss.maxcdn.com/html5shiv/3.7.2/html5shiv.min.js"></script>
	<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
	<![endif]-->
    <title>Quantitative Trait Mapping: Input File Format</title>
  </head>
  <body>
    <div class="container">
      
<nav class="navbar navbar-default">
  <div class="container-fluid">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#bs-example-navbar-collapse-1" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>

      
      

      
      <a class="navbar-brand" href="../">Home</a>

    </div>
    <div class="collapse navbar-collapse" id="bs-example-navbar-collapse-1">
      <ul class="nav navbar-nav">

	
        <li><a href="../conduct/">Code of Conduct</a></li>

	
        
        <li><a href="../setup/">Setup</a></li>
        <li class="dropdown">
          <a href="../" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-haspopup="true" aria-expanded="false">Episodes <span class="caret"></span></a>
          <ul class="dropdown-menu">
            
            <li><a href="../02-input-file-format/">Input File Format</a></li>
            
            <li><a href="../03-calc-genoprob/">Calculating Genotype Probabilities</a></li>
            
            <li><a href="../04-special-x-covar/">Special covariates for the X chromosome</a></li>
            
            <li><a href="../05-perform-genome-scan/">Performing a genome scan</a></li>
            
            <li><a href="../06-perform-perm-test/">Performing a permutation test</a></li>
            
            <li><a href="../07-find-lod-peaks/">Finding LOD peaks</a></li>
            
            <li><a href="../08-calc-kinship/">Calculating A Kinship Matrix</a></li>
            
            <li><a href="../09-perform-genome-scan-lmm/">Performing a genome scan with a linear mixed model</a></li>
            
            <li><a href="../10-perform-genome-scan-bin/">Performing a genome scan with binary traits</a></li>
            
            <li><a href="../11-est-qtl-effects/">Estimated QTL effects</a></li>
            
            <li><a href="../12-snp-assoc/">SNP association mapping</a></li>
            
            <li><a href="../13-qtl-in-do/">QTL analysis in Diversity Outbred Mice</a></li>
            
	    <li role="separator" class="divider"></li>
            <li><a href="../aio/">All in one page (Beta)</a></li>
          </ul>
        </li>
	

	
	
        <li class="dropdown">
          <a href="../" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-haspopup="true" aria-expanded="false">Extras <span class="caret"></span></a>
          <ul class="dropdown-menu">
            <li><a href="../reference/">Reference</a></li>
            
            <li><a href="../about/">About</a></li>
            
            <li><a href="../discuss/">Discussion</a></li>
            
            <li><a href="../figures/">Figures</a></li>
            
            <li><a href="../guide/">Instructor Notes</a></li>
            
          </ul>
        </li>
	

	
        <li><a href="../license/">License</a></li>
	
	
	<li><a href="/edit/gh-pages/_episodes_rmd/02-input-file-format.Rmd">Improve this page <span class="glyphicon glyphicon-pencil" aria-hidden="true"></span></a></li>
	
	
      </ul>
      <form class="navbar-form navbar-right" role="search" id="search" onsubmit="google_search(); return false;">
        <div class="form-group">
          <input type="text" id="google-search" placeholder="Search..." aria-label="Google site search">
        </div>
      </form>
    </div>
  </div>
</nav>


<div class="row">
  <div class="col-md-1">
    <h3>
      
      <a href="../"><span class="glyphicon glyphicon-menu-up" aria-hidden="true"></span><span class="sr-only">lesson home</span></a>
      
    </h3>
  </div>
  <div class="col-md-10">
    
    <h3 class="maintitle"><a href="../">Quantitative Trait Mapping</a></h3>
    
  </div>
  <div class="col-md-1">
    <h3>
      
      <a href="../03-calc-genoprob/"><span class="glyphicon glyphicon-menu-right" aria-hidden="true"></span><span class="sr-only">next episode</span></a>
      
    </h3>
  </div>
</div>

<article>
<div class="row">
  <div class="col-md-1">
  </div>
  <div class="col-md-10">
    <h1 class="maintitle">Input File Format</h1>
  </div>
  <div class="col-md-1">
  </div>
</div>


<blockquote class="objectives">
  <h2>Overview</h2>

  <div class="row">
    <div class="col-md-3">
      <strong>Teaching:</strong> 15 min
      <br/>
      <strong>Exercises:</strong> 30 min
    </div>
    <div class="col-md-9">
      <strong>Questions</strong>
      <ul>
	
	<li><p>How are the data files formatted for qtl2?</p>
</li>
	
	<li><p>Which data files are required for qtl2?</p>
</li>
	
	<li><p>Where can I find sample data for mapping with the qtl2 package?</p>
</li>
	
      </ul>
    </div>
  </div>

  <div class="row">
    <div class="col-md-3">
    </div>
    <div class="col-md-9">
      <strong>Objectives</strong>
      <ul>
	
	<li><p>To specify which input files are required for qtl2 and how they should be formatted.</p>
</li>
	
	<li><p>To locate sample data for qtl mapping.</p>
</li>
	
      </ul>
    </div>
  </div>

</blockquote>

<p>QTL mapping data consists of a set of tables of data: marker
genotypes, phenotypes, marker maps, etc. These different tables are in different comma-separated value (CSV) files. In each file, the first column is a set of IDs for the rows, and the first row is a set of IDs for the columns. For example, the genotype data file will have individual IDs in the first column, marker names for the rest of the column headers.</p>

<p><img src="../fig/iron-geno-sample.png" alt="" /></p>

<p>The sample genotype file above shows two alleles: S and B. These represent the founder strains for an intercross, which are C57BL/6 (BB) and SWR (SS) <a href="https://pubmed.ncbi.nlm.nih.gov/16799992/">(Grant et al. (2006) Hepatology 44:174-185)</a>. The B and S alleles themselves represent the haplotypes inherited from the parental strains C57BL/6 and SWR.</p>

<p>In the Diversity Outbred (DO) and Collaborative Cross (CC), alleles A to H represent haplotypes of the 8 founder strains.</p>

<p><img src="../fig/cc-founder-alleles.png" alt="" /></p>

<p>CC lines have very low heterozygosity throughout their genomes. For most loci, 
CC lines will be homozygous for one of the founder strains A-H above, and as
such will have one of only 8 genotypes (<em>e.g.</em> AA, BB, CC, …). In contrast,
DO <em>mice</em> (not lines) have high heterozygosity throughout their genomes. Each
locus will have one of 36 possible genotypes (<em>e.g.</em> AA, AB, AC, …, BB, BC, BD,…).</p>

<p><img src="../fig/cc-do-genome-comparison.png" alt="" />
For the purposes of learning QTL mapping, this lesson begins with the simplest
cases: backcrosses (2 possible genotypes) and intercrosses (3 possible genotypes).
Once we have learned how to use <code class="language-plaintext highlighter-rouge">qtl2</code> for these simpler cases, we will advance
to the most complex case involving mapping in DO mice.</p>

<p>R/qtl2 accepts the following files:</p>
<ol>
  <li>genotypes</li>
  <li>phenotypes</li>
  <li>phenotype covariates (<em>i.e.</em> tissue type, time points)</li>
  <li>genetic map</li>
  <li>physical map (optional)</li>
  <li>control file (YAML or JSON format, not CSV)</li>
</ol>

<p>We use both a genetic marker map and a physical map (if available). A sample from a genetic map of MIT markers is shown here.</p>

<p><img src="../fig/iron-geno-map-sample.png" alt="" /></p>

<p>A physical marker map provides location in bases rather than centiMorgans.</p>

<p><img src="../fig/iron-phys-map-sample.png" alt="" /></p>

<p>Numeric phenotypes are separate from the often non-numeric covariates.</p>

<p><img src="../fig/iron-pheno-sample.png" alt="" /></p>

<p>Phenotype covariates are <a href="https://en.wikipedia.org/wiki/Metadata">metadata</a> describing the phenotypes. For example, in the case of a phenotype measured over time, one column in the phenotype covariate data could be the time of measurement. For gene expression data, we would have columns representing chromosome and physical position of genes, as well as gene IDs.</p>

<p><img src="../fig/iron-phenocovar-sample.png" alt="" /></p>

<p>In addition to the set of CSV files with the primary data, we need a separate control file with various control parameters
(or metadata), including the names of all of the other data files and the genotype codes used in the genotype data file. The control file is in a specific format using either <a href="http://www.yaml.org">YAML</a> or
<a href="http://json.org">JSON</a>; these are human-readable text files for
representing relatively complex data.</p>

<p><img src="../fig/iron-control-file-sample.png" alt="" /></p>

<p>A big advantage of this control file scheme is that it greatly
simplifies the function for reading in the data. That function,
<code class="language-plaintext highlighter-rouge">read_cross2()</code>, has a <em>single</em> argument: the name (with path) of the control file.</p>

<p>For further details, see the separate <a href="http://kbroman.org/qtl2/assets/vignettes/input_files.html">vignette on the input file format</a>.</p>

<blockquote class="challenge">
  <h2 id="challenge-1">Challenge 1</h2>
  <ol>
    <li>Which data files are required by <code class="language-plaintext highlighter-rouge">qtl2</code>?</li>
    <li>Which ones are optional?</li>
    <li>How should they be formatted?</li>
  </ol>

  <blockquote class="solution">
    <h2 id="solution-to-challenge-1">Solution to Challenge 1</h2>
    <ol>
      <li>marker genotypes, phenotypes, genetic map</li>
      <li>physical map</li>
      <li>csv; JSON or YAML for control file</li>
    </ol>
  </blockquote>
</blockquote>

<h2 id="sample-data-sets">Sample data sets</h2>

<p>In this lesson, we’ll work with data sets included in the <code class="language-plaintext highlighter-rouge">qtl2</code> package. You can find out more about the <a href="http://kbroman.org/qtl2/pages/sampledata.html">sample data files</a> from the R/qtl2 web site. Zipped versions of these datasets are included with the <a href="https://github.com/rqtl/qtl2geno">qtl2geno</a> package and can be loaded into R using the <code class="language-plaintext highlighter-rouge">read_cross2()</code> function.
Additional sample data sets, including data on Diversity Outbred (DO) mice, are available at <a href="https://github.com/rqtl/qtl2data">https://github.com/rqtl/qtl2data</a>.</p>

<blockquote class="challenge">
  <h2 id="challenge-2">Challenge 2</h2>
  <p>Go to <a href="https://github.com/rqtl/qtl2data">https://github.com/rqtl/qtl2data</a> to view additional sample data.<br />
1). Find the Recla data and locate the phenotype data file. Open the file by clicking on the file name. What is in the first column? the first row?<br />
2). Locate the genotype data file, click on the file name, and view the raw data. What is in the first column? the first row?<br />
3). Locate the covariates file and open it by clicking on the file name. What kind of information does this file contain?<br />
4). Locate the control file (YAML or JSON format) and open it. What kind of information does this file contain?</p>

  <blockquote class="solution">
    <h2 id="solution-to-challenge-2">Solution to Challenge 2</h2>

    <p>1). What is in the first column of the phenotype file? Animal ID. The first row? Phenotype variable names - OF_distance_first4, OF_distance, OF_corner_pct, OF_periphery_pct, …<br />
2). What is in the first column of the genotype file? marker ID. the first row? Animal ID - 1,4,5,6,7,8,9,10, …<br />
3). Locate the covariates file and open it. What kind of information does this file contain? Animal ID, sex, cohort, group, subgroup, ngen, and coat color.<br />
4). Locate the control file (YAML or JSON format) and open it. What kind of information does this file contain? Names of primary data files, genotype and allele codes, cross type, description, and other metadata.</p>
  </blockquote>
</blockquote>

<h2 id="preparing-your-diversity-outbred-do-data-for-qtl2">Preparing your Diversity Outbred (DO) data for qtl2</h2>

<p>Karl Broman provides detailed instructions for <a href="https://kbroman.org/qtl2/pages/prep_do_data.html">preparing DO mouse data for R/qtl2</a>.</p>



<blockquote class="keypoints">
  <h2>Key Points</h2>
  <ul>
    
    <li><p>QTL mapping data consists of a set of tables of data: marker genotypes, phenotypes, marker maps, etc.</p>
</li>
    
    <li><p>These different tables are in separate comma-delimited (CSV) files.</p>
</li>
    
    <li><p>In each file, the first column is a set of IDs for the rows, and the first row is a set of IDs for the columns.</p>
</li>
    
    <li><p>In addition to primary data, a separate file with control parameters (or metadata) in either <a href="http://www.yaml.org">YAML</a> or <a href="http://json.org">JSON</a> format is required.</p>
</li>
    
    <li><p>Published and public data already formatted for QTL mapping are available on the web.</p>
</li>
    
    <li><p>These data can be used as a model for formatting your own QTL data.</p>
</li>
    
  </ul>
</blockquote>

</article>

<div class="row">
  <div class="col-md-1">
    <h3>
      
      <a href="../"><span class="glyphicon glyphicon-menu-up" aria-hidden="true"></span><span class="sr-only">lesson home</span></a>
      
    </h3>
  </div>
  <div class="col-md-10">
    
  </div>
  <div class="col-md-1">
    <h3>
      
      <a href="../03-calc-genoprob/"><span class="glyphicon glyphicon-menu-right" aria-hidden="true"></span><span class="sr-only">next episode</span></a>
      
    </h3>
  </div>
</div>


      
      
<footer>
  <div class="row">
    <div class="col-md-6" align="left">
      <h4>
	Copyright &copy; 2016–2025
	
      </h4>
    </div>
    <div class="col-md-6" align="right">
      <h4>
	
	
	<a href="/edit/gh-pages/_episodes_rmd/02-input-file-format.Rmd">Edit on GitHub</a>
	
	
	/
	<a href="/blob/gh-pages/CONTRIBUTING.md">Contributing</a>
	/
	<a href="/">Source</a>
	/
	<a href="/blob/gh-pages/CITATION">Cite</a>
	/
	<a href="mailto:susan.mcclatchy@jax.org">Contact</a>
      </h4>
    </div>
  </div>
</footer>

      
    </div>
    
<script src="../assets/js/jquery.min.js"></script>
<script src="../assets/js/bootstrap.min.js"></script>
<script src="../assets/js/lesson.js"></script>
<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
  ga('create', 'UA-37305346-2', 'auto');
  ga('send', 'pageview');
</script>

  </body>
</html>
