<!doctype html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta http-equiv="last-modified" content="2025-11-22 10:15:12 +0000">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <!-- meta "search-domain" used for google site search function google_search() -->
    <meta name="search-domain" value="">
    <link rel="stylesheet" type="text/css" href="../assets/css/bootstrap.css" />
    <link rel="stylesheet" type="text/css" href="../assets/css/bootstrap-theme.css" />
    <link rel="stylesheet" type="text/css" href="../assets/css/lesson.css" />
    
    <!-- HTML5 shim and Respond.js for IE8 support of HTML5 elements and media queries -->
    <!-- WARNING: Respond.js doesn't work if you view the page via file:// -->
    <!--[if lt IE 9]>
	<script src="https://oss.maxcdn.com/html5shiv/3.7.2/html5shiv.min.js"></script>
	<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
	<![endif]-->
    <title>Quantitative Trait Mapping: Calculating A Kinship Matrix</title>
  </head>
  <body>
    <div class="container">
      
<nav class="navbar navbar-default">
  <div class="container-fluid">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#bs-example-navbar-collapse-1" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>

      
      

      
      <a class="navbar-brand" href="../">Home</a>

    </div>
    <div class="collapse navbar-collapse" id="bs-example-navbar-collapse-1">
      <ul class="nav navbar-nav">

	
        <li><a href="../conduct/">Code of Conduct</a></li>

	
        
        <li><a href="../setup/">Setup</a></li>
        <li class="dropdown">
          <a href="../" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-haspopup="true" aria-expanded="false">Episodes <span class="caret"></span></a>
          <ul class="dropdown-menu">
            
            <li><a href="../02-input-file-format/">Input File Format</a></li>
            
            <li><a href="../03-calc-genoprob/">Calculating Genotype Probabilities</a></li>
            
            <li><a href="../04-special-x-covar/">Special covariates for the X chromosome</a></li>
            
            <li><a href="../05-perform-genome-scan/">Performing a genome scan</a></li>
            
            <li><a href="../06-perform-perm-test/">Performing a permutation test</a></li>
            
            <li><a href="../07-find-lod-peaks/">Finding LOD peaks</a></li>
            
            <li><a href="../08-calc-kinship/">Calculating A Kinship Matrix</a></li>
            
            <li><a href="../09-perform-genome-scan-lmm/">Performing a genome scan with a linear mixed model</a></li>
            
            <li><a href="../10-perform-genome-scan-bin/">Performing a genome scan with binary traits</a></li>
            
            <li><a href="../11-est-qtl-effects/">Estimated QTL effects</a></li>
            
            <li><a href="../12-snp-assoc/">SNP association mapping</a></li>
            
            <li><a href="../13-qtl-in-do/">QTL analysis in Diversity Outbred Mice</a></li>
            
	    <li role="separator" class="divider"></li>
            <li><a href="../aio/">All in one page (Beta)</a></li>
          </ul>
        </li>
	

	
	
        <li class="dropdown">
          <a href="../" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-haspopup="true" aria-expanded="false">Extras <span class="caret"></span></a>
          <ul class="dropdown-menu">
            <li><a href="../reference/">Reference</a></li>
            
            <li><a href="../about/">About</a></li>
            
            <li><a href="../discuss/">Discussion</a></li>
            
            <li><a href="../figures/">Figures</a></li>
            
            <li><a href="../guide/">Instructor Notes</a></li>
            
          </ul>
        </li>
	

	
        <li><a href="../license/">License</a></li>
	
	
	<li><a href="/edit/gh-pages/_episodes_rmd/08-calc-kinship.Rmd">Improve this page <span class="glyphicon glyphicon-pencil" aria-hidden="true"></span></a></li>
	
	
      </ul>
      <form class="navbar-form navbar-right" role="search" id="search" onsubmit="google_search(); return false;">
        <div class="form-group">
          <input type="text" id="google-search" placeholder="Search..." aria-label="Google site search">
        </div>
      </form>
    </div>
  </div>
</nav>


<div class="row">
  <div class="col-md-1">
    <h3>
      
      <a href="../07-find-lod-peaks/"><span class="glyphicon glyphicon-menu-left" aria-hidden="true"></span><span class="sr-only">previous episode</span></a>
      
    </h3>
  </div>
  <div class="col-md-10">
    
    <h3 class="maintitle"><a href="../">Quantitative Trait Mapping</a></h3>
    
  </div>
  <div class="col-md-1">
    <h3>
      
      <a href="../09-perform-genome-scan-lmm/"><span class="glyphicon glyphicon-menu-right" aria-hidden="true"></span><span class="sr-only">next episode</span></a>
      
    </h3>
  </div>
</div>

<article>
<div class="row">
  <div class="col-md-1">
  </div>
  <div class="col-md-10">
    <h1 class="maintitle">Calculating A Kinship Matrix</h1>
  </div>
  <div class="col-md-1">
  </div>
</div>


<blockquote class="objectives">
  <h2>Overview</h2>

  <div class="row">
    <div class="col-md-3">
      <strong>Teaching:</strong> 30 min
      <br/>
      <strong>Exercises:</strong> 30 min
    </div>
    <div class="col-md-9">
      <strong>Questions</strong>
      <ul>
	
	<li><p>Why would I calculate kinship between individuals?</p>
</li>
	
	<li><p>How do I calculate kinship between individuals?</p>
</li>
	
	<li><p>What does a kinship matrix look like?</p>
</li>
	
      </ul>
    </div>
  </div>

  <div class="row">
    <div class="col-md-3">
    </div>
    <div class="col-md-9">
      <strong>Objectives</strong>
      <ul>
	
	<li><p>Explain why and when kinship calculation matters in mapping.</p>
</li>
	
	<li><p>Create a kinship matrix for individuals.</p>
</li>
	
      </ul>
    </div>
  </div>

</blockquote>

<p>Population structure and kinship are common confounding factors in genome-wide association studies (GWAS), case-control studies, and other study types in genetics. They create false positive associations between genotype and phenotype at genetic markers that differ in genotype frequencies between subpopulations due to genetic relatedness between samples. Simple association tests assume statistical independence between individuals. Population structure and kinship confound associations when phenotype covariance between individuals results from genetic similarity. Accounting for relatedness between individuals helps to distinguish true associations from false positives generated by population structure or kinship.</p>

<p>As an example see the table below for phenotype and genotype frequencies between two subpopulations in a case-control study.</p>

<table>
  <thead>
    <tr>
      <th style="text-align: left"> </th>
      <th style="text-align: center">subpop1</th>
      <th style="text-align: center">subpop2</th>
      <th style="text-align: center">overall pop</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: left">frequency</td>
      <td style="text-align: center">0.5</td>
      <td style="text-align: center">0.5</td>
      <td style="text-align: center">1</td>
    </tr>
    <tr>
      <td style="text-align: left">probability of AA genotype</td>
      <td style="text-align: center">0.1</td>
      <td style="text-align: center">0.9</td>
      <td style="text-align: center">0.5</td>
    </tr>
    <tr>
      <td style="text-align: left">probability of disease</td>
      <td style="text-align: center">0.9</td>
      <td style="text-align: center">0.1</td>
      <td style="text-align: center">0.5</td>
    </tr>
    <tr>
      <td style="text-align: left">probability of disease &amp; AA</td>
      <td style="text-align: center">0.09</td>
      <td style="text-align: center">0.09</td>
      <td style="text-align: center">0.09</td>
    </tr>
  </tbody>
</table>

<p>The full population consists of two equally represented subpopulations. In the overall population, the probability of the AA genotype is 0.5, and the probability of disease is also 0.5. The joint probability of both disease and AA genotype in the population (0.09) is less than either the probability of disease (0.5) or the probability of the AA genotype (0.5) alone, and is considerably less than the joint probability of 0.25 that would be calculated if subpopulations weren’t taken into account. In a case-control study that fails to recognize subpopulations, most of the cases will come from subpopulation 1 since this subpopulation has a disease probability of 0.9. However, this subpopulation also has a low probability of the AA genotype. So a false association between AA genotype and disease would occur because only overall population probabilities would be considered.</p>

<p>Linear mixed models (LMMs) consider genome-wide similarity between all pairs of individuals to account for population structure, known kinship and unknown relatedness. They model the covariance between individuals. Linear mixed models in association mapping studies can successfully correct for genetic relatedness between individuals in a population by incorporating kinship into the model. To perform a genome scan by a linear mixed model, accounting for the relationships among individuals (in other words, including a random polygenic effect), you’ll need to calculate a kinship matrix for the individuals. This is accomplished with the <code class="language-plaintext highlighter-rouge">calc_kinship()</code> function. It takes the genotype probabilities as input.</p>

<div class="language-plaintext r highlighter-rouge"><div class="highlight"><pre class="highlight"><code>kinship &lt;- calc_kinship(probs = pr)
</code></pre></div></div>

<p>Take a look at the kinship values calculated for the first 5 individuals.</p>

<div class="language-plaintext r highlighter-rouge"><div class="highlight"><pre class="highlight"><code>kinship[1:5, 1:5]
</code></pre></div></div>

<div class="language-plaintext output highlighter-rouge"><div class="highlight"><pre class="highlight"><code>          1         2         3         4         5
1 0.6780378 0.5070425 0.4770823 0.5100762 0.5193062
2 0.5070425 0.5612483 0.5139017 0.4777593 0.5049052
3 0.4770823 0.5139017 0.7272491 0.5147456 0.5393396
4 0.5100762 0.4777593 0.5147456 0.7153455 0.5359428
5 0.5193062 0.5049052 0.5393396 0.5359428 0.5775571
</code></pre></div></div>

<p>We can also look at the first 50 mice in the kinship matrix.</p>

<div class="language-plaintext r highlighter-rouge"><div class="highlight"><pre class="highlight"><code>n_samples &lt;- 25
heatmap(kinship[1:n_samples, 1:n_samples], symm = TRUE)
</code></pre></div></div>

<div class="figure" style="text-align: center">
<img src="../fig/rmd-08-plot_kinship-1.png" alt="plot of chunk plot_kinship" />
<p class="caption">plot of chunk plot_kinship</p>
</div>

<p>The mice are listed in the same order on both sides of the matrix. The comb-like structures are called “dendrograms” and they indicate how the mice are clustered together. Each cell represents the degree of allele sharing between mice. Red colors indicate higher kinship and yellow colors indicate lower kinship. Each mouse is closely related to itself, so the cells along the diagonal tend to be darker than the other cells. You can see some evidence of related mice, possibly siblings, in the orange-shaded blocks along the diagonal.</p>

<p>By default, the genotype probabilities are converted to allele probabilities, and the kinship matrix is calculated as the proportion of shared alleles. To use genotype probabilities instead, use <code class="language-plaintext highlighter-rouge">use_allele_probs=FALSE</code> in the call to <code class="language-plaintext highlighter-rouge">calc_kinship()</code>. Further, by default we omit the X chromosome and only use the autosomes. To include the X chromosome, use <code class="language-plaintext highlighter-rouge">omit_x=FALSE</code>.</p>

<p>In calculating the kinship matrix, you can eliminate the effect of varying marker density across the genome, and only use the probabilities along the grid of pseudomarkers (defined by the <code class="language-plaintext highlighter-rouge">step</code> argument to <code class="language-plaintext highlighter-rouge">insert_pseudomarkers()</code>. To do so, we need to first use <code class="language-plaintext highlighter-rouge">calc_grid()</code> to determine the grid of pseudomarkers, and then <code class="language-plaintext highlighter-rouge">probs_to_grid()</code> to probabilities for positions that are not on the grid.</p>

<div class="language-plaintext r highlighter-rouge"><div class="highlight"><pre class="highlight"><code>grid &lt;- calc_grid(map = iron$gmap, step=1)
pr_grid &lt;- probs_to_grid(probs = pr, grid = grid)
kinship_grid &lt;- calc_kinship(probs = pr_grid)
</code></pre></div></div>

<p>On a multi-core machine, you can get some speed-up via the <code class="language-plaintext highlighter-rouge">cores</code> argument, as with <code class="language-plaintext highlighter-rouge">calc_genoprob()</code>.</p>

<div class="language-plaintext r highlighter-rouge"><div class="highlight"><pre class="highlight"><code>kinship &lt;- calc_kinship(pr, cores=4)
</code></pre></div></div>

<blockquote class="challenge">
  <h2 id="challenge-1">Challenge 1</h2>
  <p>1). Insert pseudomarkers into a new map called <code class="language-plaintext highlighter-rouge">sparser_map</code> at 
2 cM intervals.<br />
2). Calculate genotype probabilities and save as an object called <code class="language-plaintext highlighter-rouge">pr2</code>. 
Leave the error probability at the default value.<br />
3). Calculate kinship with these new probabilities. Save as <code class="language-plaintext highlighter-rouge">kinship2</code>.<br />
4). View the first several rows and columns of the <code class="language-plaintext highlighter-rouge">kinship2</code> matrix and 
compare to the original <code class="language-plaintext highlighter-rouge">kinship</code> matrix with a heatmap.</p>

  <blockquote class="solution">
    <h2 id="solution-to-challenge-1">Solution to Challenge 1</h2>

    <p>1). <code class="language-plaintext highlighter-rouge">sparser_map &lt;- insert_pseudomarkers(map = map, step = 2)</code> <br />
2). <code class="language-plaintext highlighter-rouge">pr2 &lt;- calc_genoprob(cross = iron, map = sparser_map)</code><br />
3). <code class="language-plaintext highlighter-rouge">kinship2 &lt;- calc_kinship(probs = pr2)</code><br />
4). <code class="language-plaintext highlighter-rouge">kinship2[1:5, 1:5]</code> and <code class="language-plaintext highlighter-rouge">heatmap(kinship2[1:n_samples, 1:n_samples], symm = TRUE)</code></p>
  </blockquote>
</blockquote>

<blockquote class="challenge">
  <h2 id="challenge-2">Challenge 2</h2>
  <p>Think about what a kinship matrix is and what it represents. Share your 
understanding with a neighbor. Write your explanation in the collaborative 
document or in your own personal notes.</p>

  <blockquote class="solution">
    <h2 id="solution-to-challenge-2">Solution to Challenge 2</h2>

  </blockquote>
</blockquote>


<blockquote class="keypoints">
  <h2>Key Points</h2>
  <ul>
    
    <li><p>Kinship matrices account for relationships among individuals.</p>
</li>
    
    <li><p>Kinship is calculated as the proportion of shared alleles between individuals.</p>
</li>
    
    <li><p>Kinship calculation is a precursor to a genome scan via a linear mixed model.</p>
</li>
    
  </ul>
</blockquote>

</article>

<div class="row">
  <div class="col-md-1">
    <h3>
      
      <a href="../07-find-lod-peaks/"><span class="glyphicon glyphicon-menu-left" aria-hidden="true"></span><span class="sr-only">previous episode</span></a>
      
    </h3>
  </div>
  <div class="col-md-10">
    
  </div>
  <div class="col-md-1">
    <h3>
      
      <a href="../09-perform-genome-scan-lmm/"><span class="glyphicon glyphicon-menu-right" aria-hidden="true"></span><span class="sr-only">next episode</span></a>
      
    </h3>
  </div>
</div>


      
      
<footer>
  <div class="row">
    <div class="col-md-6" align="left">
      <h4>
	Copyright &copy; 2016–2025
	
      </h4>
    </div>
    <div class="col-md-6" align="right">
      <h4>
	
	
	<a href="/edit/gh-pages/_episodes_rmd/08-calc-kinship.Rmd">Edit on GitHub</a>
	
	
	/
	<a href="/blob/gh-pages/CONTRIBUTING.md">Contributing</a>
	/
	<a href="/">Source</a>
	/
	<a href="/blob/gh-pages/CITATION">Cite</a>
	/
	<a href="mailto:susan.mcclatchy@jax.org">Contact</a>
      </h4>
    </div>
  </div>
</footer>

      
    </div>
    
<script src="../assets/js/jquery.min.js"></script>
<script src="../assets/js/bootstrap.min.js"></script>
<script src="../assets/js/lesson.js"></script>
<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
  ga('create', 'UA-37305346-2', 'auto');
  ga('send', 'pageview');
</script>

  </body>
</html>
